Add postEditor#moveSectionUp, postEditor#moveSectionDown

bantic opened the pull request:
In order to move a section that is already rendered, we have to remove it
and put a copy of it in the right place. This PR also adds a `clone` method to
sections for that purpose.

This PR removes `builder#createBlankPost`. It previously created a post with
a single empty markup section because posts couldn't have 0 sections, but that
restriction was relaxed recently. Upstream consumers of content-kit were getting
confused by having single empty markup section when booting an editor.

cc @mixonic

mixonic commented on src/js/models/card-node.js:
```diff
@@ -35,7 +35,8 @@ export default class CardNode {
       cancel: () => this.display(),
-      remove: () => this.remove()
+      remove: () => this.remove(),
+      card: this
```
`this` is the card node, not the instance of the card as a user sees it (`this.card[mode]` I believe). I'm not sure making the `CardNode` public API for card implementers is ideal.

mixonic replied:
I see how this is being used.

Could this expose the abstract node instead of the `CardNode`? The abstracts are already public or at the least "intimate" API, and unless I've read incorrectly the only other thing needed in https://github.com/bustlelabs/typeset.bustle.com/pull/21/files#diff-04c1550a2081f24687e1031a3bae973fR13 is the `editor` itself, which could be passed directly into the component.

mixonic replied:
https://github.com/bustlelabs/content-kit-editor/blob/master/CARDS.md should be updated as well.

bantic commented:
updated to add tests for all the properties that a card's `env` is expected to have, including `section` (the cardSection) instead of the `cardNode` itself. Updated the docs in CARDS.md
